src/newbank/server/NewBank.java

```diff
@@ -1,49 +1,128 @@
+        String accountName = request.substring(request.indexOf(SPACE) + 1);
+        return makeNewAccount(customer, accountName);
+      } else if (request.startsWith("MOVE")) {
+        return moveMoney(customer, request.split(SPACE));
+      }
+    }
+    return "FAIL";
+  }
+
+  private String showMyAccounts(Customer customer) {
+    return customer.accountsToString();
+  }
+
+  private String makeNewAccount(Customer customer, String accountName) {
+    customer.addAccount(new Account(accountName, 0.0));
+    customers.updateCustomer(customer);
+    return "SUCCESS";
+  }
+
+  /**
+   * The command MOVE moves the money between accounts.
+   *
+   * @param customer to which moves the moeny.
+   * @param command MOVE.
+   * @return SUCCES or FAIL
+   */
+  // TODO method is to big should be refactored
+  private String moveMoney(Customer customer, String[] command) {
+    if (customer == null) {
+      System.err.println("MOVE command: Customer is not available!");
+      return "FAIL";
+    }
+
+    // We return failure if the command doesn't match 4 arguments
+    if (command == null || command.length != 4) {
+      System.err.println("MOVE command: Only 3 arguments are expected");
+      return "FAIL";
+    }
+
+    // accessing the array directly is safe
+    // declare the variable effective final explicitely, to use them in Lambda Expression
+    final String amount = command[1];
+    final double parsedAmount;
+
+    try {
+      parsedAmount = Double.parseDouble(amount);
+    } catch (NumberFormatException nfe) {
+      System.err.printf("MOVE command: Expected number but was %s \n", amount);
+      return "FAIL";
+    }
+
+    final String from = command[2];
+    final String to = command[3];
+
+    // TODO: The from and to if detected that there are the same will return SUCCESS, we have to see
+    // if we want to allow this.
+    if (from.equals(to)) {
+      System.err.println("MOVE command: The 'from' and 'to' are the same, no move is performed!");
+      return "SUCCESS";
+    }
+
+    Set<Account> accounts = customer.getAccounts();
+    Account fromAccount = findAccount(accounts, from);
+    if (fromAccount == null) {
+      System.err.printf("MOVE command: An account with the given name: %s was not found!\n", from);
+      return "FAIL";
+    }
+
+    Account toAccount = findAccount(accounts, to);
+    if (toAccount == null) {
+      System.err.printf("MOVE command: An account with the given name: %s was not found!\n", to);
+      return "FAIL";
+    }
+
+    // Final step
+    fromAccount.setOpeningBalance(fromAccount.getOpeningBalance() - parsedAmount);
+    customer.addAccount(fromAccount);
+
+    toAccount.setOpeningBalance(toAccount.getOpeningBalance() + parsedAmount);
+    customer.addAccount(toAccount);
+
+    return "SUCCESS";
+  }
+
+  // TODO: Code should be adapted after we define an identity for class Account
+  private static Account findAccount(Set<Account> accounts, String accountName) {
```

Comment:
findAccount in NewBank class can get tricky. As NewBank class can handle multiple Accounts. instead of passing list of accounts and accountName here, can be have the same in Customer class? 

So instead of `customer.getAccounts()` , `customer.getAccount(accountName)` 

and the find logic to be implemented inside customer class ? 

Since move is a feature within Customer class, shall we have the who logic inside Customer class? 

`customer.moveMoney(fromAccount, toAccount, amount)`

Reply:
Good idea. Before any action, I'm going to wait for you and Thomas to merge the code